crates/server/Cargo.toml

```diff
@@ -53,7 +53,7 @@ dnssec = ["hickory-recursor?/dnssec"]
 recursor = ["hickory-recursor"]
 resolver = ["hickory-resolver"]
 sqlite = ["rusqlite"]
-toml = ["dep:basic-toml"]
+toml = ["dep:toml"]
```

Comment:
as I mentioned in https://github.com/hickory-dns/hickory-dns/issues/2194#issuecomment-2176497799 `basic-toml` cannot deserialize into enum variants that have fields so I switched the toml library to the `toml` crate

Reply:
IMO that should be a separate commit.